### DISCUSSION TITLE
Broken junctions will still zap players in multiplayer

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#14473] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Repairable.cs#L270 suggests that all junctions should auto-pass skill check if they have 0 voltage. Broken junctions disconnect from the power grid so I would assume they normally should have 0 voltage. 

In the sub editor, any broken or unpowered junction box won't zap you when you repair it (unless you fail the minigame)

But in multiplayer (hosting ingame), 1 out of the 8 junctions on my ship https://steamcommunity.com/sharedfiles/filedetails/?id=3312719041 (the junction marked navigation) seems to zap you even when its broken when you initially start to repair it (ie unpowered). Flipping the power switch to off doesn't seem to keep it from zapping you either if its broken. 

The others.. Don't seem to zap when broken. 

Iv noticed this behavior in beacon stations and such, so I don't think its just my submarine.

### Reproduction steps

Load bangkok submarine.

Damage junctions with explosion command set to EMP.

repair nav junction with captain, get shocked?

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by BlackMoons** [comment]
Did some more testing and it also effects the engine junction on the bangkok. Apparently it was the 'power consumption' value in the other junctions 'power transfer' component that causes it to properly power down when damaged. Most vanilla junctions don't have a non zero power consumption value in the power transfer component so likely affects most/all vanilla junctions.

**Comment by BlackMoons** [comment]
Upon some code inspection, https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Power/Powered.cs#L160

Might be at fault. I think it should return back 0 voltage if 'isBroken' is true?

**Comment by Regalis11** [comment]
This has been fixed in our private repo now, fix coming up in a future update!
